src/test/java/com/crud/tasks/controller/TaskControllerTest.java

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    @Test
+    public void testDeleteTask() throws Exception {
+
+        //given
+        Task task = new Task(
+                1L,
+                "test task",
+                "test to do"
+        );
+
+        when(service.getTask(task.getId())).thenReturn(Optional.of(task));
+        doNothing().when(service).deleteTask(task.getId());
+
+        //when & then
+        mockMvc.perform(delete("/task/tasks/{id}", task.getId())
+                .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+                .andExpect(status().isNoContent()
+                );
+        verify(service, times(1)).getTask(task.getId());
+        verify(service, times(1)).deleteTask(task.getId());
+        verifyNoMoreInteractions(service);
+    }
+
+    @Test
+    public void testUpdateTask() throws Exception {
+
+        //given
+        Task task = new Task(
+                1L,
+                "test task",
+                "test to do"
+        );
+
+        TaskDto taskDto = new TaskDto(
+                1L,
+                "test task",
+                "test to do"
+        );
+
+        when(taskMapper.mapToTask(any(TaskDto.class))).thenReturn(task);
+        when(service.saveTask(any(Task.class))).thenReturn(task);
+        when(taskMapper.mapToTaskDto(any(Task.class))).thenReturn(taskDto);
+
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(taskDto);
+
+        //when & then
+        mockMvc.perform(put("/task/tasks/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(("UTF-8"))
+                .content(jsonContent))
+                .andExpect(status().is(200))
+                .andExpect(jsonPath("$.id", is(1)))
+                .andExpect(jsonPath("$.title", is("test task")))
+                .andExpect(jsonPath("$.content", is("test to do"))
+                );
+        verify(taskMapper, times(1)).mapToTask(any(TaskDto.class));
+        verify(taskMapper, times(1)).mapToTaskDto(any(Task.class));
+        verify(service, times(1)).saveTask(task);
+        verifyNoMoreInteractions(taskMapper);
+        verifyNoMoreInteractions(service);
+    }
+
+    @Test
+    public void shouldCreateTask() throws Exception {
+
+        //given
+        Task task = new Task(
+                1L,
+                "test task",
+                "test to do"
+        );
+
+        TaskDto taskDto = new TaskDto(
+                1L,
+                "test task",
+                "test to do"
+        );
+
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(taskDto);
+
+        when(taskMapper.mapToTask(taskDto)).thenReturn(task);
+        when(service.saveTask(task)).thenReturn(task);
+
+        //when & then
+        mockMvc.perform(post("/task/tasks")
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(("UTF-8"))
+                .content(jsonContent))
+                .andExpect(status().is(201))
```

Comment:
minor: it is more readable when you use constants, like HttpStatus.CREATED.value()